x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectRealm.java

```diff
@@ -385,6 +381,15 @@ public void close() {
         openIdConnectAuthenticator.close();
     }
 
+    /*
+     * We only map claims that are of Type String, Boolean, or Number, or arrays that contain only these types
+     */
+    private static boolean isAllowedTypeForClaim(Object o) {
+        return (o instanceof String || o instanceof Boolean || o instanceof Number
+            || (o instanceof Collection && ((Collection) o).stream()
+            .allMatch(c -> c instanceof String || c instanceof Boolean || c instanceof Number)));
```

Comment:
I think we probably want to ensure that these are heterogenous collections. I don't know how we would make sense of `[ "1", true, 1.0 ]`

Reply:
> think we probably want to ensure that these are heterogenous collections. 

You used Greek so I get to be pedantic :tada: s/heterogeneous/homogeneous/ :)
In all seriousness, I took a lenient stance here on purpose for metadata ( as opposed to user properties where I allowed only strings ). 

> I don't know how we would make sense of [ "1", true, 1.0 ]

- Metadata can be used as metadata, for informational purposes so we do not necessarily need to make sense of the value of a given key.
- It would work in a role mapping rule such as 
```
"rules" : { "field" : { "groups" : "1" } },
```

happy to reconsider if you think this would cause an issue somewhere else or that it is too lenient for the sake of being lenient but I don't see any specific problem with it as is

Reply:
My comment clearly says `homoeneous` 😈 

Fair enough - as long as it's a considered position, I'm not worried.